iceoryx_hoofs/include/iceoryx_hoofs/internal/posix_wrapper/unix_domain_socket.hpp

```diff
@@ -98,7 +98,7 @@ class UnixDomainSocket : public DesignPattern::Creation<UnixDomainSocket, IpcCha
 
     /// @brief checks whether the unix domain socket is outdated
     /// @return true if the unix domain socket is outdated, false otherwise, IpcChannelError if error occured
-    cxx::expected<bool, IpcChannelError> isOutdated() noexcept;
+    static cxx::expected<bool, IpcChannelError> isOutdated() noexcept;
```

Comment:
@elBoberido I would disagree. `isOutdated` does not access the object at all and can be therefore static.
But you can also call it like it would be a member function. `mySocket.isOutdated()` works fine.

I think this is a consequence of our ugly IPC design which has to be refactored.

Reply:
@elfenpiff I disagree here, even if is a bad API it doesn't make it better if there is a static `ìsOutdated` function without any parameter. The IPC channel currently requires such a method and if one starts using `Foo::isOutdated()` and later on changes the using declaration to a message queue, the code breaks. We need to fulfill the contract and the contract says it is a method.

Reply:
@elBoberido I think the contract states that `Foo.isOutdated()` is compilable.
But we can make it unstatic again :D

Reply:
@elfenpiff sure but that would then break the API and you know Hyrum's Law, even if it doesn't make sense, there will be developer using it

Reply:
@elBoberido @elfenpiff 
I removed the static here and in the other occurences since it is dangerous to use when the underlying IPC Channel changes.